src/OpenTelemetry.Extensions.Serilog/OpenTelemetrySerilogSink.cs

```diff
@@ -0,0 +1,110 @@
+
+            this.openTelemetryLoggerProvider = openTelemetryLoggerProvider!;
+            this.disposeProvider = disposeProvider;
+
+            var logEmitter = this.openTelemetryLoggerProvider.CreateEmitter();
+            Debug.Assert(logEmitter != null, "logEmitter was null");
+
+            this.logEmitter = logEmitter!;
+
+            // TODO: This project can only access IncludeFormattedMessage
+            // because it can see SDK internals. At some point this is likely
+            // not to be the case. Need to figure out where to put
+            // IncludeFormattedMessage so that extensions can see it. Ideas:
+            // Make it public on OpenTelemetryLoggerProvider or expose it on
+            // LogEmitter instance.
+            this.includeFormattedMessage = this.openTelemetryLoggerProvider.IncludeFormattedMessage;
+        }
+
+        public void Emit(LogEvent logEvent)
+        {
+            Debug.Assert(logEvent != null, "LogEvent was null.");
+
+            LogRecordData data = new(Activity.Current)
+            {
+                Timestamp = logEvent!.Timestamp.UtcDateTime,
+                LogLevel = (LogLevel)(int)logEvent.Level,
+                Message = this.includeFormattedMessage ? logEvent.RenderMessage() : logEvent.MessageTemplate.Text,
+                Exception = logEvent.Exception,
+            };
+
+            LogRecordAttributeList attributes = default;
+            foreach (KeyValuePair<string, LogEventPropertyValue> property in logEvent.Properties)
+            {
+                // TODO: Serilog supports complex type logging. This is not yet
+                // supported in OpenTelemetry.
+                if (property.Key == Constants.SourceContextPropertyName
+                    && property.Value is ScalarValue sourceContextValue)
+                {
+                    data.CategoryName = sourceContextValue.Value as string;
+                }
+                else if (property.Value is ScalarValue scalarValue)
+                {
+                    attributes.Add(property.Key, scalarValue.Value);
+                }
+                else if (property.Value is SequenceValue sequenceValue)
+                {
+                    IReadOnlyList<LogEventPropertyValue> elements = sequenceValue.Elements;
+                    if (elements.Count > 0 && elements[0] is ScalarValue)
+                    {
+                        object[] values = new object[elements.Count];
+                        for (int i = 0; i < elements.Count; i++)
+                        {
+                            if (elements[i] is ScalarValue value)
+                            {
+                                values[i] = value.Value;
+                            }
+                        }
+
+                        attributes.Add(property.Key, values);
+                    }
+                }
+            }
+
+            this.logEmitter.Log(in data, in attributes);
```

Comment:
wondering if it was intentional that we called it "Log" instead of "Emit" ?

Reply:
I thought "Log" as a name was more idiomatic for a .NET API but I don't feel strongly about it. Also considered "Write" 🤷

Reply:
I vote for Emit because:
1. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/logging-library-sdk.md#logemitter
2. It communicates to the users that this is not a normal Logging API that they should be using.

Reply:
Renamed